str/catalog_design/polymorphic_site_extractor.py

```diff
@@ -0,0 +1,157 @@
+    --output-dir "hoptan-str/catalog-design/" \
+    --memory=32G --storage=20G \
+    polymorphic_site_extractor.py --vcf-path=gs://cpg-tob-wgs-test/hoptan-str/shard_workflow_test/merge_str_vcf_combiner/combined_eh.vcf \
+    --catalog-path=gs://cpg-tob-wgs-test/hoptan-str/5M_run/5M_sharded_100k/chunk_1.json
+    --folder_name=sharded_polymorphic_catalog
+
+"""
+import json
+import hail as hl
+import click
+
+
+from cpg_utils import to_path
+from cpg_utils.hail_batch import output_path, init_batch
+
+
+def polymorphic_site_extractor(file_path):
+    """Extracts polymorphic sites from a VCF file and returns a list of REPIDs (similar to rsids) representing those sites
+    The output has been benchmarked with Gymrek's statSTR and agrees with the number of polymorphic sites found
+    """
+    init_batch()
+    # read in VCF into mt format
+    mt = hl.import_vcf(file_path)
+
+    mt = hl.sample_qc(mt)
+    mt = hl.variant_qc(mt)
+
+    # remove chrY and chrM
+    filtered_mt = mt.filter_rows(~hl.str(mt.locus.contig).startswith('chrY'))
+    filtered_mt = filtered_mt.filter_rows(
+        ~hl.str(filtered_mt.locus.contig).startswith('chrM')
+    )
+
+    # remove loci that are monomorphic for the REF allele
+    filtered_mt = filtered_mt.filter_rows(hl.len(filtered_mt.alleles) > 1)
+
+    # at a biallelic locus, remove loci that are monomorphic for 1 ALT allele
+    filtered_mt = filtered_mt.filter_rows(
+        ~(
+            (hl.len(filtered_mt.variant_qc.AC) == 2)
+            & (filtered_mt.variant_qc.AC[0] == 0)
+        )
+    )
+
+    # collect the REPIDs into one list
+    rep_id_list = filtered_mt.info.REPID.collect()
+
+    return rep_id_list
+
+
+def catalog_filter(rep_id_list, catalog_path, gcs_output_path):
+    """Retains loci in a JSON file that intersect with a list of REPIDs (STR equiv. of rsids) and writes the filtered catalog to a new JSON file"""
+    with to_path(catalog_path).open('r') as json_file:
+        # Load the JSON content
+        catalog = json.load(json_file)
+
+    filtered_data = []
+
+    # Make the rep_id_list into a set
+    polymorphic_rep_id_set = set(rep_id_list)
+
+    for entry in catalog:
+        # Check if 'VariantId' exists, use 'LocusId' otherwise
+        entry_variant_ids = (
+            set(entry['VariantId']) if 'VariantId' in entry else {entry['LocusId']}
+        )
+
+        # Append to filtered_data if the entry Variant Id intersects with polymorphic_rep_id_set
+        if entry_variant_ids & polymorphic_rep_id_set:
+            filtered_data.append(entry)
+
+    # Write to output
+    with to_path(gcs_output_path).open('w') as out_file:
+        json.dump(filtered_data, out_file, indent=2)
+    return filtered_data
+
+
+def catalog_sharder(filtered_data, chunk_size, folder_name):
+    """Shards a filtered catalog JSON file into chunks of size chunk_size"""
+    if not isinstance(filtered_data, list):
+        print('Invalid JSON format. The file should contain a list.')
+        return
```

Comment:
Is this the behaviour you want? If this isn't a valid data type you're returning and carrying on instead of quitting. This might be a good candidate for an exception being raised as this method doesn't return anything

```suggestion
    if not isinstance(filtered_data, list):
        raise ValueError('Invalid JSON format. The file should contain a list.')
```